fxapom/pages/sign_in.py

```diff
@@ -2,79 +2,85 @@
-                        EC.visibility_of_element_located(self._email_input_locator))
-                    self._sign_in_window_handle = handle
-                    break
-            else:
-                raise Exception('Popup has not loaded')
-        else:
-            self.popup = False
+    @property
+    def login_password(self):
+        """Get the value of the login password field."""
+        return self.driver.find_element(*self._password_input_locator).get_attribute('value')
 
-    def sign_in(self, email, password):
-        """Signs in using the specified email address and password."""
-        self.email = email
-        if self.is_element_visible(*self._next_button_locator):
-            self.click_next()
-        self.login_password = password
-        self.click_sign_in()
+    @login_password.setter
+    def login_password(self, value):
+        """Set the value of the login password field."""
+        password = self.driver.find_element(*self._password_input_locator)
+        password.clear()
+        password.send_keys(value)
+
+    def click_next(self):
+        self.driver.find_element(*self._next_button_locator).click()
 
     @property
     def email(self):
         """Get the value of the email field."""
-        return self.selenium.find_element(*self._email_input_locator).get_attribute('value')
+        return self.driver.find_element(*self._email_locator).get_attribute('value')
 
     @email.setter
     def email(self, value):
         """Set the value of the email field."""
-        email = Wait(self.selenium, self.timeout).until(
+        email = Wait(self.driver, self.timeout).until(
             EC.visibility_of_element_located(self._email_input_locator))
         email.clear()
         email.send_keys(value)
 
-    @property
-    def login_password(self):
-        """Get the value of the login password field."""
-        return self.selenium.find_element(*self._password_input_locator).get_attribute('value')
-
-    @login_password.setter
-    def login_password(self, value):
-        """Set the value of the login password field."""
-        password = self.selenium.find_element(*self._password_input_locator)
-        password.clear()
-        password.send_keys(value)
-
-    def click_next(self):
-        self.selenium.find_element(*self._next_button_locator).click()
+    def check_for_popup(self, handles):
+        if len(handles) > 1:
+            self.popup = True
+            for handle in handles:
+                self.driver.switch_to.window(handle)
+                if self.is_element_visible(*self._fox_logo_locator):
+                    Wait(self.driver, self.timeout).until(
+                        EC.visibility_of_element_located(self._email_input_locator))
+                    self._sign_in_window_handle = handle
+                    break
+            else:
+                raise Exception('Popup has not loaded')
+        else:
+            self.popup = False
```

Comment:
Nit: If you set `self.popup` to `False` initially, you wouldn't need these two lines.